### ISSUE TITLE
DISCO training fails silently

### ISSUE THREAD

**Issue by JulienVig** [epfml/disco#607]
**Clicking on the "Train" button in the pre-defined tasks of the DISCO frontend, both in local and on the [website](https://epfml.github.io/disco/#/) doesn't start the training.**

The error is only triggered when using the web client, local tests all pass.

1. Go to any pre-defined tasks such as the [MNIST task](https://epfml.github.io/disco/#/mnist), either on the website or on your local instance running the `develop` branch
2. Click on 'NEXT' two times until reaching the training step. (Uploading data doesn't change the outcome)
3. Click on 'Train Alone' or 'Train collaboratively'
4. Inspect element and check the console logs:
```
TypeError: this.datasetBuilder.isBuilt is not a function
    startTraining index.js:95
    Lc index.js:7
    Lc index.js:3
    startTraining index.js:93
    0 templateLoader.js:29
    d runtime-core.esm-bundler.js:155
    p runtime-core.esm-bundler.js:164
    t runtime-dom.esm-bundler.js:369
```

**Expected behavior**
Clicking on the Train button should start the training successfully or return an error message to the user specifying there is a problem, such message could be: "You didn't uplaod any data" or "The service is currently unavailable".

**Screenshots**
<img width="1094" alt="Screen Shot 2024-01-11 at 16 43 50" src="https://github.com/epfml/disco/assets/33122365/b76ca7d6-4fda-480a-b2ec-757f7d7aba34">

**Desktop (please complete the following information):**
 - MacOS
 - Firefox

**Comment by JulienVig** [comment]
Edit: the error arises both on the website and on a local web-client running the `develop`branch

**Comment by JulienVig** [comment]
Update: uploading data before training triggers the same error

**Comment by JulienVig** [comment]
[This commit](https://github.com/epfml/disco/commit/6e85f6f1f8b772e15c5e03cb5b55ef4030db1894#diff-04761da9988636b38bf48b80fb0fcab0a5b47c7bba0254dc48233159441c8613) changes the method `dataset_builder.isBuilt()` into the getter function `get built()` without updating the frontend call to `datasetBuilder.isBuilt()` in `Trainer.vue`.
Getter function calls don't use parentheses, so I changed the call to `datasetBuilder.built` in `Trainer.vue. I found the same mistake for `datasetBuilder.size`, we should be on the lookout for others.
